src/lib/score/graphics/widgets/QGraphicsRangeSlider.cpp

```diff
@@ -0,0 +1,138 @@
+#include <score/graphics/widgets/QGraphicsRangeSlider.hpp>
+#include <score/model/Skin.hpp>
+
+#include <QGraphicsSceneMouseEvent>
+#include <QPainter>
+
+#include <wobjectimpl.h>
+W_OBJECT_IMPL(score::QGraphicsRangeSlider);
+
+namespace score
+{
+QGraphicsRangeSlider::QGraphicsRangeSlider(QGraphicsItem* parent)
+{
+  auto& skin = score::Skin::instance();
+  setCursor(skin.CursorPointingHand);
+}
+
+void QGraphicsRangeSlider::setStart(double start)
+{
+  if (m_start != start)
+  {
+    m_start = std::clamp(start, 0., 1.);
+    update();
+  }
+}
+
+void QGraphicsRangeSlider::setEnd(double end)
+{
+  if (m_end != end)
+  {
+    m_end = std::clamp(end, 0., 1.);
+    update();
+  }
+}
+
+void QGraphicsRangeSlider::setValue(ossia::vec2f value)
+{
+  setStart(value[0]);
+  setEnd(value[1]);
+}
+
+ossia::vec2f QGraphicsRangeSlider::value() const noexcept
+{
+  return {float(m_start), float(m_end)};
+}
+
+void QGraphicsRangeSlider::setRange(double min, double max)
+{
+  {
+    m_min = min;
+    m_max = max;
+    update();
+  }
+}
+
+void QGraphicsRangeSlider::mousePressEvent(QGraphicsSceneMouseEvent* event)
+{
+  if (event->button() == Qt::LeftButton)
```

Comment:
add `using namespace std;` at the top of functions which use abs (or math functions in general)